### ISSUE TITLE
Exception while updating route line

### ISSUE THREAD

**Issue by cafesilencio** [mapbox/mapbox-navigation-android#2880]
When testing the `examples` test app run into the following crash

```
04-30 20:39:44.153 32563-599/com.mapbox.navigation.examples E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-2
    Process: com.mapbox.navigation.examples, PID: 32563
    java.lang.IllegalStateException: Calling getLayer when a newer style is loading/has loaded.
        at com.mapbox.mapboxsdk.maps.Style.validateState(Style.java:597)
        at com.mapbox.mapboxsdk.maps.Style.getLayer(Style.java:238)
        at com.mapbox.navigation.ui.route.MapRouteLine$decorateRouteWithCongestionIndicators$1.invokeSuspend(MapRouteLine.kt:503)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

Was creating different routes quickly and closing / re-opening the `OffboardRouterActivity` before getting the response. In other words, stressing it :trollface:

There was a leak 👇 previous to the crash.

![offboard_router_activity_map_route_line_leak](https://user-images.githubusercontent.com/1668582/80821224-51b6fc00-8ba6-11ea-832d-4b96ceda824b.png)

cc @cafesilencio

**Comment by Guardiola31337** [comment]
I was able to repro OP after #2882 🤔 cc @cafesilencio @langsmith

**Comment by cafesilencio** [comment]
I believe this issue was resolved with PR #2899
